hs_app_netCDF/models.py

```diff
@@ -342,6 +342,7 @@ def get_hs_term_dict(self):
         return hs_term_dict
 
     def update_netcdf_file(self, user):
+        self.metadata.refresh_from_db()
```

Comment:
why did you add the refresh into here?

Reply:
This ensures that we get the updated value of the `metadata.is_dirty` property (which might get set due to changes to metadata elements) on which the update of netcdf file depends. This was not needed before since the call to `self.metadata` used to retrieve the metadata object from DB each time such a call was made.

Reply:
Also see my comment here: https://github.com/hydroshare/hydroshare/pull/3212/files#diff-0daa45deebe2e6282237d0c4e20456c9R710